### ISSUE TITLE
Jackhammer ups power consumption

### ISSUE THREAD

**Issue by archigo** [CleverRaven/Cataclysm-DDA#31754]
# Describe the bug

The electric jackhammer, with 7000 charges, can only mine around two solid rock walls before going empty. Meanwhile the gasoline powered one, with 400 charges, only spent 10 charges mining a solid rock wall

# Steps To Reproduce

1. Spawn an UPS powered jackhammer
2. Mine Solid rock walls

# Expected behavior

It makes sense that the electric jackhammer does not last as long as the gasoline one, but only being able to mine two walls is absurd, considering how much time and power it takes to recharge it. The equivalent of 200 charges from the gasoline powered jackhammer is what i would consider appropriate.

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1809
- Game Version: 0.D-4883-g0db43cf [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food]
]

**Comment by archigo** [comment]
Currently electric jackhammer is set to spend 3500 charges. I suggest it be set to 500 instead. This would allow 14 usages of a fully charged electric jackhammer, vs the 20 usages of a full gasoline jackhammer.

**Comment by Zireael07** [comment]
We recently had a discussion on it (did you search for jackhammer before making an issue), and jackhammers take A LOT of power. I mean A LOT. If anything, gasoline version needs charges used amped up.

**Comment by TechyBen** [comment]
Yeah, usually electric is much less use efficient than other fuel sources due to power densities.

**Comment by archigo** [comment]
I see the issue that made it consume 3500 charges. That discussion didn't consider how long it takes to recharge a jackhammer though. As it is now, you can use it twice and then you have to go do something else while it recharges in the recharge station. Of course you get slightly more uses if you have really good bionics or carrying a lot of UPS. From a gameplay perspective, the current jackhammer is just not usable, and there is no reason whatsoever to consider it over the gasoline powered one, since gasoline is easy to get. Additionally there is no downtime to reloading the gasoline jackhammer.

In the real world you would have quick replacement batteries that are charging while you work,, so you would not have downtime. In the game we don't really have rechargeable batteries though, so it becomes an unwieldy tool that is not really usefull.

A better solution in my opinion, would be to make is use small or medium storage batteries as magazines, which you could have a couple of with you and some charging in the car.

Perhaps an even better solution, and more generally usefull to all the electronics in the game would be to create a cable drum. We already have the jumper cables, but their range is too limited for these kind of used, and also don't go though z levels i believe.

As it is right now though, i believe it would be appropriate to reduce charges used into the 500-1000 range.

**Comment by TechyBen** [comment]
> In the real world you would have quick replacement batteries that are charging while you work,, so you would not have downtime. 

IRL you would neve power it with a battery.

Might need some balancing. But it does not need to be the same as the gasoline one.

However, a "jumper cable" that works with items in your inventory/welded sounds interesting too. (As an addition to the current use case of the existing item)2

**Comment by kevingranade** [comment]
How long it takes to charge it is irrelevant, the power density is what matters.

I've used portable electric jackhammers (usually called hammerdrills because the primary use case is an enhanced drill, not a percussive tool), and you would NEVER use one of these for a job like taking down a rock or concrete wall. Not only would it run out of power constantly, but it would also burn itself out in short order with the kind of duty cycles required when bashing a wall into pieces.

In other words, if you TRIED to use one for this use case, you'd see exactly what you are seeing in the game, it's a really terrible tool for this use case and it ends up being really bad at it.

> However, a "jumper cable" that works with items in your inventory/welded sounds interesting too.

I'm not sure how this is different from a UPS.

**Comment by TechyBen** [comment]
@kevingranade Oh, as in, connection to a generator. UPS would also work, but it would be interesting gameplay mechanic possibly for a jumper cable to run from say a generator or car battery.

I understand that would need coding. I recently sold (neve used it myself) a compressor unit and a load of compressor/pneumatic tools. I sold a generator with a tool hookup (for powering just about any type of power tool, but with both consumer sockets, and industrial type rated/protected sockets). Might be out of the scope for the game, but it is clever the type of IRL heavy duty equipment there is. :)
